smartsim/_core/launcher/dragon/dragonBackend.py

```diff
@@ -69,12 +74,33 @@ class ProcessGroupInfo:
     puids: t.Optional[t.List[t.Optional[int]]] = None  # puids can be None
     return_codes: t.Optional[t.List[int]] = None
     hosts: t.List[str] = field(default_factory=list)
+    redir_workers: t.Optional[ProcessGroup] = None
 
     @property
     def smartsim_info(self) -> t.Tuple[SmartSimStatus, t.Optional[t.List[int]]]:
         return (self.status, self.return_codes)
 
 
+# Thanks to Colin Wahl from HPE HPC Dragon Team
+def redir_worker(io_conn: Connection, file_path: str) -> None:
+    """Read stdout/stderr from the Dragon connection.
+
+    :param io_conn: Dragon connection to stdout or stderr
+    :type io_conn: Connection
+    :param file_path: path to file to write to
+    :type file_path: str
+    """
+    try:
+        with open(file_path, "a", encoding="utf-8") as file_to_write:
+            while True:
+                output = io_conn.recv()
+                print(output, flush=True, file=file_to_write, end="")
```

Comment:
logger?

Reply:
In this case we want to use print (to file) just to redirect the process stdout/stderr